draft-ietf-quic-http.md

```diff
@@ -331,18 +339,40 @@ pushes via the SETTINGS_ENABLE_PUSH setting in the SETTINGS frame (see
 {{connection-establishment}}), which is disabled by default.
 
 As with server push for HTTP/2, the server initiates a server push by sending a
-PUSH_PROMISE frame containing the Stream ID of the stream to be pushed, as well
-as request header fields attributed to the request. The PUSH_PROMISE frame is
-sent on the stream of the associated (client-initiated) request, while the
-Promised Stream ID field specifies the Stream ID of the server-initiated
-request.
+PUSH_PROMISE frame that includes request header fields attributed to the
+request. The PUSH_PROMISE frame is sent on a response stream.  Unlike HTTP/2,
+the PUSH_PROMISE does not reference a stream; when a server fulfills a promise,
+the stream that carries the stream headers references the PUSH_PROMISE.  This
+allows a server to fulfill promises in the order that best suits its needs.
+
+The server push response is conveyed on a push stream.  A push stream is a
+server-initiated stream.  A push stream includes a header (see
+{{fig-push-stream-header}}) that identifies the PUSH_PROMISE that it fulfills.
+This header consists of a 32-bit Push ID, which identifies a server push (see
+{{frame-push-promise}}).
+
+~~~~~
+ 0                   1                   2                   3
+ 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                         Push ID (32)                          |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+~~~~~
+{: #fig-push-stream-header title="Push Stream Header"}
+
+Each Push ID MUST only be used once in a push stream header.  If a push stream
+header includes a Push ID that was used in another push stream header, the
+client MUST treat this as a connection error of type HTTP_DUPLICATE_PUSH.  The
+same Push ID can be used in multiple PUSH_PROMISE frames (see
+{{frame-push-promise}}).
 
-The server push response is conveyed in the same way as a non-server-push
-response, with response headers and (if present) trailers carried by HEADERS
-frames, and response body (if any) carried by DATA frames.
+After the push stream header, a push contains a response ({{request-response}}),
+with response headers and (if present) trailers carried by HEADERS frames sent
+on the control stream, and response body (if any) carried by DATA frames.
```

Comment:
Heh, you noticed that, but then you didn't notice that the ordering was completely wrong :)